x/auth/ante/basic.go

```diff
@@ -92,7 +93,16 @@ func (cgts ConsumeTxSizeGasDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 	}
 	params := cgts.ak.GetParams(ctx)
 
-	ctx.GasMeter().ConsumeGas(params.TxSizeCostPerByte*sdk.Gas(len(ctx.TxBytes())), "txSize")
+	// charge extra gas for authz/MsgGrantAuthorization msgs.
+	extraGasCost := sdk.Gas(0)
+	txLen := len(ctx.TxBytes())
+	for _, msg := range tx.GetMsgs() {
+		if msg.Type() == authz.TypeMsgGrantAuthorization {
+			extraGasCost += params.TxSizeCostPerByte * uint64(txLen/len(tx.GetMsgs()))
```

Comment:
How will this be known client side? Unless a user simulates the transaction it will fail. Out of scope of this PR but we should provide an endpoint that displays all the custom gas amount.

Reply:
Yeah, I agree with @marbar3778 here, this is way too hacky and not scalable

maybe tomorrow's discussion will give more clarity on an overall protocol base fee to avoid msg pollution, and we don't even need this PR anymore?